includes/class-freemius.php

```diff
@@ -4174,13 +4174,17 @@ function has_api_connectivity( $flush_if_no_connectivity = false ) {
                 ! empty( $this->_storage->connectivity_test ) &&
                 isset( $this->_storage->connectivity_test['is_active'] )
             ) {
-                $is_active = $this->_storage->connectivity_test['is_active'];
+                $is_connected = isset( $this->_storage->connectivity_test['is_connected'] ) ?
+                    $this->_storage->connectivity_test['is_connected'] :
+                    null;
+                $is_active    = ( $this->_storage->connectivity_test['is_active'] || is_object( $this->_site ) );
             } else {
-                $is_active = $this->should_turn_fs_on( $this->apply_filters( 'is_plugin_update', $this->is_plugin_update() ) );
-
-                $this->store_connectivity_info( (object) array( 'is_active' => $is_active ), null );
+                $is_connected = null;
+                $is_active    = $this->should_turn_fs_on( $this->apply_filters( 'is_plugin_update', $this->is_plugin_update() ) );
             }
 
+            $this->store_connectivity_info( (object) array( 'is_active' => $is_active ), $is_connected );

```

Comment:
@fajardoleo A question, would it work if we do it like this instead

```php
if ( $is_active ) {
  $this->_is_on = true;
  $this->update_connectivity_info( $is_connected );
}
```

Maybe it will also save us to not cache the connectivity in cache in case it could not be determined?

Reply:
@swashata Looks great! By the way, since `$this->store_connectivity_info()` updates `$this->_has_api_connection`, it will look like this:
```
if ( $is_active ) {
    $this->_is_on = true;
    $this->update_connectivity_info( $is_connected );
} else if ( $is_connected ) {
    $this->_has_api_connection = true;
}

return $this->_has_api_connection;
```

But please check to see if there's an alternative to the `else if` part. Thanks!